### PR TITLE
Basic plots to track allocation usage

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR makes an endpoint to make some basic plots to track allocation usage.

### CODE REVIEW

File: skyportal/handlers/api/allocation.py [hunk 1]
```diff
@@ -222,3 +228,139 @@ def delete(self, allocation_id):
         self.verify_and_commit()
         self.push_all(action='skyportal/REFRESH_ALLOCATIONS')
         return self.success()
+
+
+class AllocationAnalysisHandler(BaseHandler):
```
Comment (guynir42) [inline skyportal/handlers/api/allocation.py:233]: Maybe `AllocationReportHandler` makes it more distinct from "analysis" which is a whole other thing?

File: skyportal/handlers/api/allocation.py [hunk 2]
```diff
@@ -222,3 +228,139 @@ def delete(self, allocation_id):
...
+        def make_autopct_observations(values):
+            def my_autopct(pct):
+                total = sum(values)
+                if np.isnan(pct):
+                    val = 0
+                else:
+                    val = int(round(pct * total / 100.0))
+                return f'{pct:.0f}%  ({val:d} Observations)'
+
+            return my_autopct
```
Comment (guynir42) [inline skyportal/handlers/api/allocation.py:314]: Is there an easy way to not duplicate all these functions?

**Review by guynir42 — APPROVED** [review]
Minor comments, but looks good.

File: skyportal/handlers/api/allocation.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/allocation.py]: Yes good point.

File: skyportal/handlers/api/allocation.py (hunk 2 above)
Comment (mcoughlin) [reply skyportal/handlers/api/allocation.py]: Done.

**Pull request merged**